p2p/subscriber.go

```diff
@@ -100,23 +110,34 @@ func (s *Subscriber[H]) SetVerifier(val func(context.Context, H) error) error {
 			log.Errorw("invalid header",
 				"from", p.ShortString(),
 				"err", err)
+			s.metrics.observeReject(ctx)
 			return pubsub.ValidationReject
 		}
-		// keep the valid header in the msg so Subscriptions can access it without
-		// additional unmarhalling
-		msg.ValidatorData = hdr
 
 		var verErr *header.VerifyError
 		err = val(ctx, hdr)
 		switch {
-		case err == nil:
-			return pubsub.ValidationAccept
-		case errors.As(err, &verErr) && verErr.SoftFailure:
-			return pubsub.ValidationIgnore
 		default:
+			s.metrics.observeReject(ctx)
 			return pubsub.ValidationReject
+		case errors.As(err, &verErr) && verErr.SoftFailure:
+			s.metrics.observeIgnore(ctx)
+			return pubsub.ValidationIgnore
+		case err == nil:
+		}
+
+		now := time.Now()
+		if !lastAccept.IsZero() {
+			s.metrics.observeAccept(ctx, now.Sub(lastAccept), len(msg.Data))
 		}
```

Comment:
I don't think we we want this. because the moment when we call SetVerifier is not representative of propagation time.

Reply:
Yeah, I agree. Looks like declaring lastAccept inside callback func would do the trick. Instead of capturing single value in closure manner like it is now.

Reply:
Inside pval? no, then the lastAccept would then be local to the callback and zero on each new verification attempt

Reply:
Ah I see, you want to observe intervals between Accept results. One more option is to store it inside metrics struct and access inside of accept() observation calls. It will allow you to move observation code away from main decision making logic